src/Components/DynamicFields.res

```diff
@@ -24,8 +29,12 @@ let make = (~paymentType, ~list, ~paymentMethod, ~paymentMethodType, ~setRequire
     )
     ->Js.Dict.fromArray
     ->Js.Json.object_
-  } else {
+  } else if (
+    PaymentMethodsRecord.dynamicFieldsEnabledPaymentMethods->Js.Array2.includes(paymentMethodType)
```

Comment:
Inside PaymentMethodsRecord the check was actually on the fieldsArr, which is used to render the input fields. However, in the confirm payload the Dynamic Fields were still being passed for the payment methods which are not present in the arrayList

Reply:
Shouldn't the check below still be there for fieldsArr? Its still needed for rendering right?

Reply:
Rendering the fields is based on concatenation of Dynamic fields and paymentMethodsFields in **getPaymentMethodFields** which we store in the fieldsArr. Since for non enabled payment methods Dynamic Fields will be Js.Json.null, it will only pick from paymentMethodsFields.